entity-framework/core/miscellaneous/configuring-dbcontext.md

```diff
@@ -155,6 +155,36 @@ using (var context = serviceProvider.GetService<BloggingContext>())
 
 var options = serviceProvider.GetService<DbContextOptions<BloggingContext>>();
 ```
+## Avoiding DbContext threading issues
+
+Entity Framework Core does not support multiple parallel operations being run on the same `DbContext`. Because of this, it's important to use separate `DbContext` instances for operations that may execute in parallel. Not doing so could result in invalid operation exceptions.
+
+### Avoiding DbContext threading issues with async methods
+
+One code pattern that could lead to parallel operations with a single `DbContext` is the use of asynchronous extension methods. These methods enable Entity Framework Core operations to be performed in a non-blocking way, but if callers do not `await` completion of one operation before starting another, a `DbContext` could encounter issues from running more than one operation in parallel. You should be sure to `await` the completion of any asynchronous operation before beginning another with the same `DbContext`.
+
+### Avoiding DbContext threading issues with dependency injection
```

Comment:
IMO this whole section boils down to “don’t use DbContext in parallel”, and DI usage seems to have very little incidence.

One way DI misuse could lead to concurrency is if you made the DbContext a singleton. But that can be generalized as any pattern that leads to sharing the same DbContext instance across multiple requests. But this and singleton instances seem like they can be explained without even mentioning DI.

Reply:
I agree. I included the DI discussion, in particular, since it's one we've seen a couple customers have questions about. Do you think the doc would be better if I shortened this part up?